### PR TITLE
added build_analytics function

### PR DESCRIPTION
**Pull request by rkrashr**

### CODE REVIEW

**Comment by yhagerjs** [comment]
Was this tested? I don't think it would work as is..

File: SonarExternal.cmake [hunk 1]
```diff
@@ -1147,3 +1147,23 @@ function(build_cpp_redis)
   target_include_external_directory(cpp_redis::lib cpp_redis source_dir includes)
   target_include_external_directory(cpp_redis::lib cpp_redis source_dir tacopie/includes)
 endfunction()
+
+function(build_analytics)
+  # builds analytics as an external java project
+  # version is not used for now as library is new
```
Comment (yhagerjs) [inline SonarExternal.cmake]: it's better to use a version. git tag also works. This way you have no idea what you're installing, and you can't build old versions reliably.

File: SonarExternal.cmake [hunk 2]
```diff
@@ -1147,3 +1147,23 @@ function(build_cpp_redis)
...
+  ExternalProject_Add(analytics
+    GIT_REPOSITORY https://github.com/jsonar/analytics.git
+    DOWNLOAD_NO_PROGRESS 1
+    CMAKE_ARGS
+      -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+      -DCMAKE_INSTALL_PREFIX=
+      -DCMAKE_POSITION_INDEPENDENT_CODE=ON
+    INSTALL_COMMAND DESTDIR=<INSTALL_DIR> ${CMAKE_MAKE_PROGRAM} install
+    )
+  sonar_external_project_dirs(analytics install_dir)
```
Comment (yhagerjs) [inline SonarExternal.cmake]: Are you referencing the built project somewhere?

File: SonarExternal.cmake [hunk 3]
```diff
@@ -1147,3 +1147,23 @@ function(build_cpp_redis)
...
+  if (NOT ANALYTICS_VERSION)
+    set(ANALYTICS_VERSION 1.0.2)
+  endif()
+  message(STATUS "Building analytics-${ANALYTICS_VERSION}")
+  ExternalProject_Add(analytics
+    GIT_REPOSITORY https://github.com/jsonar/analytics.git
+    DOWNLOAD_NO_PROGRESS 1
+    CMAKE_ARGS
+      -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+      -DCMAKE_INSTALL_PREFIX=
```
Comment (yhagerjs) [inline SonarExternal.cmake]: you probably want `-DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>`

File: SonarExternal.cmake [hunk 4]
```diff
@@ -1147,3 +1147,23 @@ function(build_cpp_redis)
...
+      -DCMAKE_POSITION_INDEPENDENT_CODE=ON
+    INSTALL_COMMAND DESTDIR=<INSTALL_DIR> ${CMAKE_MAKE_PROGRAM} install
```
Comment (yhagerjs) [inline SonarExternal.cmake]: there's no need for `INSTALL_COMMAND`

File: SonarExternal.cmake [hunk 5]
```diff
@@ -1147,3 +1147,23 @@ function(build_cpp_redis)
...
+function(build_analytics)
+  # builds analytics as an external java project
+  # version is not used for now as library is new
+  cmake_parse_arguments(ANALYTICS "" "VERSION" "" ${ARGN})
+  if (NOT ANALYTICS_VERSION)
+    set(ANALYTICS_VERSION 1.0.2)
+  endif()
+  message(STATUS "Building analytics-${ANALYTICS_VERSION}")
+  ExternalProject_Add(analytics
+    GIT_REPOSITORY https://github.com/jsonar/analytics.git
```
Comment (yhagerjs) [inline SonarExternal.cmake]: This won't work without a username and password

**Comment by rkrashr** [comment]
Why do you think so?
Looks like it is working.. But again, may be there are some hidden issues which I am not aware of.
Do you want to explain?

File: SonarExternal.cmake (hunk 5 above)
Comment (rkrashr) [reply SonarExternal.cmake]: Is there a way to provide credentials?
Buildbot pulls from git anyway..

Comment (yhagerjs) [reply SonarExternal.cmake]: `GIT_REPOSITORY https://jsonar-buildbot:m7wSYJTLd6ZWYtDCY3NeZzC+WLo1FUirSG1DkXXqbQo=@github.com/jsonar/analytics.git`

**Comment by yhagerjs** [comment]
because I can't see how the buildbot has access to the repo without credentials.

File: SonarExternal.cmake (hunk 2 above)
Comment (rkrashr) [reply SonarExternal.cmake]: it is used by maven and it handles the artifact

File: SonarExternal.cmake (hunk 3 above)
Comment (rkrashr) [reply SonarExternal.cmake]: it fails without that line, trying to copy cmake eula.txt elsewhere

File: SonarExternal.cmake (hunk 4 above)
Comment (rkrashr) [reply SonarExternal.cmake]: same thing,
it fails without that line, trying to copy cmake eula.txt elsewhere

**Comment by yhagerjs** [comment]
This seems stale by now. Please reopen if you still need this.

**Pull request closed without merging**